### PR TITLE
Set up env vars on app entry

### LINKED ISSUE #1225: We should run `getEnv()` on entry to the application
**Is your feature request related to a problem? Please describe.**
Currently, the logic which switches between using local, test or live environment values is triggered in the `server/index`. This means environment variables cannot be used before `server/index` is run. We need to change this so that environment variables are set up on entry to the application. 

**Describe the solution you'd like**
Move the `getEnv()` trigger to `src/index.js` which is the entry point to the application. This is a simple refactor of the code execution order.

**Describe alternatives you've considered**
N/A

**Testing notes**

Dev insight: Regression test the application -- Ensure the correct environment variables continue to work both via 'npm start' and 'npm run dev'

**Additional context**
This is blocking the logging work which requires environment variables. 

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by pjlee11**
Resolves #1225

**Overall change:** _Ensures that the APP_ENV is used to set up the environment config variables as the first thing in the application. Therefore picking which `.env` to use. Also adds documentation about the entry points to the application._

**Code changes:**

- Refactors the `getEnv/index.js` logic so that the initial setting of the config variables is performed on import of the file. Therefore removing this logic from the `getEnv()`. This is to differentiate between the use of this file in `src/index.js` and in `webpack.config.js`
- Move the logic for setting up which `.env` file from `src/server/index.jsx` to use into `src/index.js` because the latter is the application entry point. 
- Moves unneeded variables (`APP_ENV`, `DOT_ENV_CONFIG`) in `webpack.config.js` into `webpack.config.client.js` where they are actually needed. 
- Updates the tests so that the `src/server/env/index.js` can be required multiple times without being cached, this is achieved by `jest.resetModules()`.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [x] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: docs/Developer-Help.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Developer help
+
+This is a wiki for helping developers working in this application. It has detailed explanations of the application an known gotchas. 
+
+## Code execution overview
+
+Webpack bundles the `src/` code into `build/server.js` which is what is actually executed via `npm run start`. The following describes the code execution order prior to webpack bundling the `src/` code.
```
Comment (dr3) [inline docs/Developer-Help.md]: >Webpack bundles the `src/` code into `build/server.js`

Thats not exactly true, it uses an entry point to bundle _most_ of `src/`

File: docs/Developer-Help.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Developer help
+
+This is a wiki for helping developers working in this application. It has detailed explanations of the application an known gotchas.
```
Comment (sareh) [inline docs/Developer-Help.md]:
```suggestion
This is a wiki for helping developers working in this application. It has detailed explanations of the application and known gotchas. 
```

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import http from 'http';
+// import and instantly use 'process.env.APP_ENV' to define the correct `.env` file to use
```
Comment (sareh) [inline src/index.js]: It'd be better if this explanation of what `/server/env` does was at the top of that file instead.

**Comment by bcmn** [comment]
> Therefore picking which .env to use.

Not sure this is accurate -- the bake-script does this.

**Review by dr3 — APPROVED** [review]

**Comment by pjlee11** [comment]
@bcmn as spoken about in person, when running `npm run build` on local or Jenkins we pass in the `process.env.APP_ENV` which is used in `/src/server/env/index.js` to decided which `.env` file to use for the bundling process. 

TL;DR
`APP_ENV=test webpack`  = create an application bundle using `.env.test`

**Review by bcmn — APPROVED** [review]
Approving, based on the understanding we plan to remove the reuse of `getEnv()` between webpack & runtime in a follow-up PR. I would rather the build-time logic live only inside the webpack client, & `getEnv` was repurposed to only serve the run-time requirements.

**Review by sareh — APPROVED** [review]

**Comment by jamesbrumpton** [comment]
LGTM
